src/threading/Manager.cc

```diff
@@ -36,6 +38,127 @@ Manager::~Manager() {
+                                           auto* s = get_message_thread_stats();
+                                           prometheus::ClientMetric metric;
+                                           metric.gauge.value = static_cast<double>(s->sent_out_total);
+                                           return metric;
+                                       });
+
+    pending_messages_in_metric =
+        telemetry_mgr->GaugeInstance("zeek", "msgthreads_pending_in_messages", {}, "Pending number of inbound messages",
+                                     "", []() -> prometheus::ClientMetric {
+                                         auto* s = get_message_thread_stats();
+                                         prometheus::ClientMetric metric;
+                                         metric.gauge.value = static_cast<double>(s->pending_in_total);
+                                         return metric;
+                                     });
+    pending_messages_out_metric =
+        telemetry_mgr->GaugeInstance("zeek", "msgthreads_pending_out_messages", {},
+                                     "Pending number of outbound messages", "", []() -> prometheus::ClientMetric {
+                                         auto* s = get_message_thread_stats();
+                                         prometheus::ClientMetric metric;
+                                         metric.gauge.value = static_cast<double>(s->pending_out_total);
+                                         return metric;
+                                     });
+
+    pending_message_in_buckets_fam =
+        telemetry_mgr->GaugeFamily("zeek", "msgthreads_pending_messages_in_buckets", {"lt"},
+                                   "Number of threads with pending inbound messages split into buckets");
+    pending_message_out_buckets_fam =
+        telemetry_mgr->GaugeFamily("zeek", "msgthreads_pending_messages_out_buckets", {"lt"},
+                                   "Number of threads with pending outbound messages split into buckets");
+
+    for ( auto idx : pending_bucket_brackets ) {
+        std::string idx_str;
+        if ( idx == std::numeric_limits<uint64_t>::infinity() )
+            idx_str = "inf";
+        else
+            idx_str = std::to_string(idx);
+
+        current_bucketed_messages.pending_in[idx] = 0;
+        current_bucketed_messages.pending_out[idx] = 0;
+
+        pending_message_in_buckets[idx] =
+            pending_message_in_buckets_fam->GetOrAdd({{"lt", idx_str}}, [idx]() -> prometheus::ClientMetric {
+                auto* s = get_message_thread_stats();
+                prometheus::ClientMetric metric;
+                metric.gauge.value = static_cast<double>(s->pending_in.at(idx));
+                return metric;
+            });
+        pending_message_out_buckets[idx] =
+            pending_message_out_buckets_fam->GetOrAdd({{"lt", idx_str}}, [idx]() -> prometheus::ClientMetric {
+                auto* s = get_message_thread_stats();
```

Comment:
Nothing to do: I just thought that a callback that'n not bound to a given counter/gauge would work nicely here.